calliope/analysis/plotting/energy_flows.py

```diff
@@ -0,0 +1,319 @@
+
+Plot energy flows data.
+"""
+
+import pandas as pd
+
+from calliope.analysis.plotting.util import break_name, get_range
+
+
+def _line(loc_coordinates, location, carrier, tech, prod, scale_factor,
+          techs_colors, is_initial_timestep, add_legend, name):
+    [transmission_type, from_location] = tech.split(":")
+    hover_info = "%s>%s: %.2f by %s (%s)" % \
+        (from_location, location, prod, transmission_type, carrier)
+    # e.g. "Region1>Region2: 256.54 by gas_transmission (gas)"
+
+    line = dict(
+        visible=False,
+        mode="lines",
+        hoverinfo="text",
+        text="",
+        line=dict(
+            width=prod * scale_factor + 1,
+            color=techs_colors[transmission_type]
+        ),
+        legendgroup=transmission_type,
+        opacity=0.6,
+        showlegend=False
+    )
+
+    line_legend = dict(
+        visible=False,
+        mode="lines",
+        hoverinfo="text",
+        text="",
+        line=dict(
+            width=10,
+            color=techs_colors[transmission_type]
+        ),
+        legendgroup=transmission_type,
+        name=break_name(name, 18),
+        opacity=0.6,
+    )
+
+    line_info_marker = dict(
+        visible=False,
+        mode="markers",
+        hoverinfo="text",
+        text=hover_info,
+        marker=dict(
+            symbol="square",
+            opacity=0,
+            color=techs_colors[transmission_type]
+        ),
+        legendgroup=transmission_type,
+        name=tech,
+        showlegend=False
+    )
+
+    if set(loc_coordinates.index) == set(["x", "y"]):
+        h_coord, v_coord = "x", "y"
+    elif set(loc_coordinates.index) == set(["lon", "lat"]):
+        h_coord, v_coord = "lon", "lat"
+
+    line[h_coord] = [
+        loc_coordinates[from_location][h_coord],
+        loc_coordinates[location][h_coord]
+    ]
+    line[v_coord] = [
+        loc_coordinates[from_location][v_coord],
+        loc_coordinates[location][v_coord]
+    ]
+    line_legend[h_coord] = [None]
+    line_legend[v_coord] = [None]
+    line_info_marker[h_coord] = [(1 / 2) * (loc_coordinates[from_location][h_coord] +
+                                 loc_coordinates[location][h_coord])]
+    line_info_marker[v_coord] = [(1 / 2) * (loc_coordinates[from_location][v_coord] +
+                                 loc_coordinates[location][v_coord])]
+
+    if is_initial_timestep:
+        # plot only the first timestep data when the chart is initialized
+        line["visible"] = True
+        line_legend["visible"] = True
+        line_info_marker["visible"] = True
+    if add_legend:
+        return [line, line_legend, line_info_marker]
+    else:
+        return [line, line_info_marker]
+
+
+def _marker(loc_coordinates, location, carrier, tech, prod, scale_factor,
+            techs_colors, is_initial_timestep, add_legend, name):
+    hover_info = "%s: %.2f of %s (%s)" % \
+        (location, prod, tech, carrier)
+    # ex: "Region1: 3552.65 of pipe_import (gas)"
```

Comment:
Can you move the comment to above the line it is commenting on, and expand to ``Example hover_info: "Region1: ..."``